src/backends/ncnn/ncnnlib.cc

```diff
@@ -0,0 +1,232 @@
+/**
+ * DeepDetect
+ * Copyright (c) 2018 Jolibrain
+ * Author: Corentin Barreau <corentin.barreau@epitech.eu>
+ *
+ * This file is part of deepdetect.
+ *
+ * deepdetect is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * deepdetect is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with deepdetect.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "outputconnectorstrategy.h"
+#include <thread>
+
+// NCNN
+#include "ncnnlib.h"
+#include "ncnninputconns.h"
+#include "cpu.h"
+#include "net.h"
+
+namespace dd
+{
+    int my_hardware_concurrency()
+    {
+        std::ifstream cpuinfo("/proc/cpuinfo");
+
+        return std::count(std::istream_iterator<std::string>(cpuinfo),
+			  std::istream_iterator<std::string>(),
+			  std::string("processor"));
+    }
+  
+    unsigned int hardware_concurrency()
+    {
+        unsigned int cores = std::thread::hardware_concurrency();
+        if (!cores)
+            cores = my_hardware_concurrency();
+        if (cores > 4)
```

Comment:
This is hacky, let's remove it. I suggest we add a `threads` parameter to the `mllib` object at service creation. If unset, the `hardware_concurrency` value applies, otherwise the specified `threads` value applies.